packages/engine/crawlers/solana/token.ts

```diff
@@ -55,3 +65,51 @@ export const getSPLTokenDocument = async (
 		metadata: metadata,
 	};
 };
+
+export const solanaFungiblesByAddress = async (
+	context: SolanaContext,
+	ownerPubkey: PublicKey,
+) => {
+	const { connection } = context;
+	const accountKeys = [ownerPubkey];
+	const fungiblePromises: Promise<TokenDocument>[] = [
+		getNativeTokenDocument(context, ownerPubkey),
+	];
+
+	watchLogs(context, ownerPubkey);
+
+	const parsedTokenAccounts = await throttle(() => {
+		return connection.getParsedTokenAccountsByOwner(
+			ownerPubkey,
+			tokenProgramFilter,
+			'confirmed',
+		);
+	})();
+
+	for (const { pubkey, account } of parsedTokenAccounts.value) {
+		const tokenType = getTokenType(account);
+
+		if (tokenType === TokenType.Fungible) {
+			fungiblePromises.push(
+				getSPLTokenDocument(context, pubkey, account, ownerPubkey),
+			);
+		}
+
+		accountKeys.push(pubkey);
+	}
+
+	const fungibleTokens = (await Promise.all(fungiblePromises)).filter(
+		(token) => token.account.mint === solMint || token.account.balance !== '0',
+	);
```

Comment:
this filter out the tokens which balance = 0 in `fungibleTokens`, but in other chain like tezos, we continue to display tokens with 0 balance, do we need to unify this UI issue? @tanlethanh